src/vmm/src/rpc_interface.rs

```diff
@@ -333,16 +344,9 @@ impl<'a> PrebootApiController<'a> {
 
         // Init the data store from file, if present.
         if let Some(data) = metadata_json {
-            vm_resources
-                .locked_mmds_or_default()
-                .put_data(
-                    serde_json::from_str(data)
-                        .expect("MMDS error: metadata provided not valid json"),
-                )
-                .map_err(|err| {
-                    error!("Populating MMDS from file failed: {:?}", err);
-                    crate::FcExitCode::GenericError
-                })?;
+            vm_resources.locked_mmds_or_default().put_data(
+                serde_json::from_str(data).expect("MMDS error: metadata provided not valid json"),
+            )?;
```

Comment:
This change removes the `error!("Populating MMDS from file failed: {:?}", err);` log message.

This should be more explicitly documented.

Reply:
The log message is still there, its just implicit now. Its part of the error types `Display` implementation, so we get it when when `main` logs the error. We have a test that verifies this exact log message's presence, which still passes

Reply:
But this doesn't appear as a log anymore, right? as in:

```
2023-11-23T14:55:39.305207904 [anonymous-instance:main] Populating MMDS from file failed ...
```

I'm not sure whether this is a problem, though.

Reply:
It appears as an `error!` log, with slightly different formatting (because it also prints all the encapsulating errors now):
```
 2023-11-23T13:19:32.079476801 [0c631bdd-4617-46d8-a62f-c68da695bce7:main] RunWithApiError error: Failed to build MicroVM: Populating MMDS from file failed: ...
```

Reply:
oh ok! awesome. I thought `.expect` would bypass the logger.

Reply:
The `.expect` was there before, rustfmt just moved it :)

Reply:
If we have a test that covers this I am satisfied with this (also considering we have previously noted we do not uphold the same stringent stability guarantees for log messages).